src/main/java/io/prometheus/cloudwatch/CloudWatchCollector.java

```diff
@@ -699,6 +687,34 @@ private void scrape(List<MetricFamilySamples> mfs) {
       samples.add(new MetricFamilySamples.Sample(
           "cloudwatch_exporter_scrape_error", new ArrayList<>(), new ArrayList<>(), error));
       mfs.add(new MetricFamilySamples("cloudwatch_exporter_scrape_error", Type.GAUGE, "Non-zero if this scrape failed.", samples));
+
+      String buildVersion = "";
+      String releaseDate = "";
+      int errorFlag = 0;
+      try {
+          final Properties properties = new Properties();
+          properties.load(CloudWatchCollector.class.getClassLoader().getResourceAsStream(".properties"));
+          buildVersion = properties.getProperty("BuildVersion");
+          releaseDate = properties.getProperty("ReleaseDate");
```

Comment:
The build info, by its nature, never changes. Could we do this work up-front and only set the metric once, or otherwise store the values to avoid unnecessary per-scrape work?

Reply:
Yes I can move it up, and stop committing to the master before its ready to be looked at again! Sorry

Reply:
All good, I was just passing through and couldn't keep from commenting 😬. Let me know when you want another round!